### DISCUSSION TITLE
sccahe in gitlab ci - only cache misses

### DISCUSSION THREAD

**Discussion by samopolacek** [mozilla/sccache#1829] [Q&A]
I am trying to integrate sccache into our gitlab pipelines.

My test job runs build twice with build dir removal in between the runs. The first run fills in local sccache cache and the second build is then able to use the cache and is considerably faster. 

When I create gitlab cache from the sccache local cache, run the build only once in the job, and run the job multiple times, all I get is cache misses. Absolute path to build directory is static.

I tried to enable traces for sccache to investigate what's going on and I think that the reason is that environment variables are different between the jobs, e.g. CI_PIPELINE_CREATED_AT (timestamp).

I assume that the same would be true for gitlab pipelines, however, I found no special treatment of environment variables anywhere that would suggest that environment has to be the same between builds.

What am I not overlooking?

**Comment by sylvestre** [comment]
I moved this into a conversation as it isn't currently an actionable issue. Sorry
